### PR TITLE
Add function to error with enabled isolation

### PR DESCRIPTION
**Pull request by christianpoveda**
Fixes https://github.com/rust-lang/miri/issues/986
r? @RalfJung

### CODE REVIEW

File: src/helpers.rs [hunk 1]
```diff
@@ -304,4 +304,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
         self.eval_libc(name)?.to_i32()
     }
+
+    fn avoid_isolation(&mut self, name: &str) -> InterpResult<'tcx> {
```
Comment (RalfJung) [inline src/helpers.rs]: "avoid" is somewhat weak. Maybe `check_no_isolation`?

File: src/helpers.rs [hunk 2]
```diff
@@ -304,4 +304,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
...
+        if !self.eval_context_mut().machine.communicate {
+            throw_unsup_format!("`{}` not available when isolation is enabled", name)
```
Comment (RalfJung) [inline src/helpers.rs]: Please add a remark like "Pass -Zmiri-disable-isolation to disable it."

File: src/helpers.rs (hunk 1 above)
Comment (christianpoveda) [reply src/helpers.rs]: I'm renaming this to `banish_isolation` :P

**Comment by RalfJung** [comment]
Looking good!

Please rebase across https://github.com/rust-lang/miri/pull/975 once that landed to also use this for the clock shims.

**Comment by christianpoveda** [comment]
OK we are good to go

**Comment by RalfJung** [comment]
Thanks! @bors r+

**Pull request merged**